src/modules/twap/utils/getTwapOrderStatus.test.ts

```diff
@@ -0,0 +1,70 @@
+import { getTwapOrderStatus } from './getTwapOrderStatus'
+
+import { TwapOrdersExecution } from '../hooks/useTwapOrdersExecutions'
+import { TwapOrderStatus, TWAPOrderStruct } from '../types'
+
+const orderStruct: TWAPOrderStruct = {
+  sellToken: '0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6',
+  buyToken: '0x02ABBDbAaa7b1BB64B5c878f7ac17f8DDa169532',
+  receiver: '0xe9B79591E270B3bCd0CC7e84f7B7De74BA3D0E2F',
+  partSellAmount: '0xc4c6e2ce9530cf',
+  minPartLimit: '0x091f5db5e0561ce4',
+  t0: 0,
+  n: 4,
+  t: 300,
+  span: 0,
+  appData: '0x226123123d9e99f7e8b46519b6311acac9f3d5e661b1fb40dc638eae03e156df',
+}
+
+describe('getTwapOrderStatus()', () => {
+  describe('When executedSellAmount equals to partSellAmount * n', () => {
+    it('Then an order status is Fulfilled', () => {
+      const execution: TwapOrdersExecution = {
+        confirmedPartsCount: orderStruct.n,
+        info: {
+          executedSellAmount: (BigInt(orderStruct.partSellAmount) * BigInt(orderStruct.n)).toString(), // <----
```

Comment:
I assume, since all parts are fill-or-kill, it's not possible, is it?

Reply:
I also checked with fully filled TWAP order

Reply:
> I assume, since all parts are fill-or-kill, it's not possible, is it?

Technically yes.
But I've seen this happening.
Although I might be confusing with limit orders...

Reply:
Then I have no idea how to fix it now :(

Reply:
There is already code in the project that deals with this:

https://github.com/cowprotocol/cowswap/blob/0ae6602f707045079c8a5e0f3b8d6bd70f703e61/src/utils/orderUtils/isOrderFilled.ts#L28

Be aware that your solution should work most of the time, I'm raising an edge case.